### DISCUSSION TITLE
STM32F1 without OS hangs at tusb_fifo:_ff_push()

### DISCUSSION THREAD

**Discussion by mme-motorsport** [hathach/tinyusb#996] [General]
Hello,

I'm having issues with STM32F105 and F103 with hanging at the tusb_fifo:_ff_push().

`
// send one item to FIFO WITHOUT updating write pointer
static inline void _ff_push(tu_fifo_t* f, void const * app_buf, uint16_t rel)
{
  memcpy(f->buffer + (rel * f->item_size), app_buf, f->item_size);
}
`

It hard faults at the memcpy. I was not able to find the cause. I did find that f->item_size is 0x800 so i'm assuming it just tries to copy something that is too small to copy, but I can't seem to find where this buffer gets allocated. 

Does anyone have tinyusb running successfully on stm32f1 that could share a light? Code from the examples (blue pill) doesn't change anything.

Board is working correctly, if I use the default STM32CubeMX example, device initializes correctly.

Thanks

**Comment by hathach** [comment]
could be same reason as https://github.com/hathach/tinyusb/discussions/990

**Comment by duempel** [comment]
@mme-motorsport could you provide more information about your setup? I'm using STM32F105 for more than one year with TinyUSB and never run into an issue like this. Just tested the current master and cdc_msc demo. Hard fault within memcpy seems like an unaligned access. Which IDE and compiler are you using?
